### PR TITLE
Es6

### PR DESCRIPTION
**Pull request by simobasso**
move whole package to es6

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        loader: 'ng-annotate'
+      }, {
+        test: /\.js$/,
+        loader: 'babel',
+        exclude: /node_modules/,
+        query: {
+          presets: [
+            "latest",
+          ],
+          "plugins": [ "istanbul" ]
```
Comment (dennybiasiolli) [inline webpack.config.js]: please don't use double quotes in property name

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    failOnHint: false,
+  },
+  eslint: {
+    parserOptions: {
+      ecmaVersion: 6,
+      sourceType: "module"
+    },
+    emitError: false,
+    failOnError: false,
+    "rules": {
```
Comment (dennybiasiolli) [inline webpack.config.js]: please remove this rule

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    validateIndentation: 2,
+    emitErrors: false,
+    failOnHint: false,
+  },
+  eslint: {
+    parserOptions: {
+      ecmaVersion: 6,
+      sourceType: "module"
+    },
+    emitError: false,
```
Comment (dennybiasiolli) [inline webpack.config.js]: please set true on emitError and failOnError

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            "latest",
+          ],
+          "plugins": [ "istanbul" ]
+        }
+      }
+    ],
+  },
+  jscs: {
+    validateIndentation: 2,
+    emitErrors: false,
```
Comment (dennybiasiolli) [inline webpack.config.js]: please set true on emitError and failOnHint

File: src/angular-pdf.directive.js [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      let debug = false;
+      let url = scope.pdfUrl;
+      let httpHeaders = scope.httpHeaders;
+      let pdfDoc = null;
+      let pageToDisplay = isFinite(attrs.page) ? parseInt(attrs.page) : 1;
+      let pageFit = attrs.scale === 'page-fit';
+      let scale = attrs.scale > 0 ? attrs.scale : 1;
+      let canvasid = attrs.canvasid || 'pdf-canvas';
+      let canvas = $document[0].getElementById(canvasid);
+      var creds = attrs.usecredentials;
```
Comment (dennybiasiolli) [inline src/angular-pdf.directive.js]: can we use `let` also for `creds`, `ctx` and `windowEl`?

File: package.json [hunk 6]
```diff
@@ -1,21 +1,42 @@
...
+  "bugs": {
+    "url": "https://github.com/sayanee/angular-pdf/issues"
   },
   "keywords": [
     "pdf",
     "angular",
     "directive"
   ],
-  "license": "MIT",
+  "dependencies": {
```
Comment (dennybiasiolli) [inline package.json]: Please check if we can use `peerDependencies` instead of `dependencies`

**Comment by dennybiasiolli** [comment]
@simobasso can we also have an `angular-pdf.js` file in `dist` folder after building with webpack, like in the current master?

**Comment by rhyskoedijk** [comment]
Could I make a small suggestion since you are already writing this.
Consider exposing the PDF.js `_pdfDoc` in the `onLoad` callback, which would allow end-users to provide their own hooks in to the document. I wanted to explore the adding "find text" functionality to my viewer but as Angular-PDF keeps the document internal, it is impossible to extend currently.

> if (angular.isFunction(scope.onLoad)) {
>     scope.onLoad(**_pdfDoc**);
> }

Maybe also a new callback at the start of the `renderPage` function to allow end-users to perform their own extra logic/rendering too?

**Comment by rhyskoedijk** [comment]
Can I fork `simobasso:es6` and submit my own pull requests for this new version, or is that just making things overly complicated? I am interested in:

- Switching the `canvasId` to a `canvasClassName`.  Class lookup rather than id looking works better when using multiple instances on the same page since you cannot generated unique canvas ids using angular bindings since the `link` function runs before any bindings in the template are resolved.
- Exposing `pdfDoc` in the `onLoad` callback as an extension point for end-users
- Exposing an `onRender` callback as an extension point for end-users

An issue I had around the `pdfTask` being cancelled when using multiple instances seems to be resolved in this version already as you have re-scoped the variables within the directive.

**Comment by dennybiasiolli** [comment]
Thanks for the feedback @rhyskoedijk, this speeds up our testing-before-merging process 😉 

Our roadmap on the project is the following:

- merge this PR in order to provide a better structured project

- improve the project with the new features suggested

Starting from tomorrow we will fix conflicts in this PR in order to merge as soon as possible, so you can fork the `master` branch directly.
Your suggestions are great, after PR's merge we will glad to focus our attention to them and if you would like to contribute with PRs we can improve this directive together 😄 

Thanks for your help

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    new webpack.optimize.UglifyJsPlugin()
+  )
+}
+
+config.output = isTest ? {} : {
+  path: __dirname + '/dist',
+  publicPath: isProd ? '/' : 'http://localhost:8080/',
+  filename: "angular-pdf.min.js",
+  library: "pdf",
+  libraryTarger: "umd",
```
Comment (dennybiasiolli) [inline webpack.config.js]: @simobasso I think this property is `libraryTarget` instead of `libraryTarger`

File: package.json [hunk 8]
```diff
@@ -1,46 +1,66 @@
...
+    "phantomjs-prebuilt": "^2.1.4",
+    "webpack": "^1.14.0",
+    "webpack-dev-server": "^1.16.2"
   },
   "scripts": {
-    "bower": "bower",
-    "build": "grunt",
-    "start": "http-server ./example -p 8081",
-    "test": "grunt --verbose"
+    "build": "rimraf dist && webpack --progress --profile",
```
Comment (dennybiasiolli) [inline package.json]: @simobasso can we install `rimraf` as devDependency?

**Comment by dennybiasiolli** [comment]
@simobasso what do you think about [this branch](https://github.com/sayanee/angularjs-pdf/tree/es6)?
It resolves conflicts, fixes sintax errors and provide both standard and minimized output.

**Comment by rhyskoedijk** [comment]
just merge it ;)

**Comment by simobasso** [comment]
close in favor of #184

**Pull request closed without merging**